### PR TITLE
[Block/DataFrame] check for unique column names, ...

### PR DESCRIPTION
**Pull request by jgrewe**
added respective test

### CODE REVIEW

File: include/nix/Block.hpp [hunk 1]
```diff
@@ -446,12 +446,21 @@ class NIXAPI Block : public base::EntityWithMetadata<base::IBlock> {
...
                 std::string msg = "Incompatible DataType for column ";
                 throw std::invalid_argument(msg + c.name);
             }
         }
+        std::sort(names.begin(), names.end());
+        names.erase(std::unique(names.begin(), names.end()), names.end());
+        if (names.size() != cols.size()) {
+            throw ConsistencyError("Block::createDataFrame: Column names must be unique!");
+        }
+
```
Comment (gicmo) [inline include/nix/Block.hpp:463]: I am ok with this but wouldn't using a [`set`](https://en.cppreference.com/w/cpp/container/set) be more convenient?

Comment (jgrewe) [reply include/nix/Block.hpp]: Yes, of course, thanks for pointing me to it!

**Comment by gicmo** [comment]
Submission to codecov worked ([logs](https://travis-ci.org/github/G-Node/nix/jobs/680693447#L868) but codecov says its waiting for CI.

**Comment by achilleas-k** [comment]
Force merging.

**Pull request merged**